test/new-e2e/tests/agent-metric-logs/log-agent/utils/file_tailing_utils.go

```diff
@@ -36,91 +37,129 @@ type LogsTestSuite interface {
 }
 
 // AppendLog append log with 'content', which is then repeated 'reccurrence' times and verifies log contents.
-func AppendLog(ls LogsTestSuite, logFile, content string, recurrence int) {
+func AppendLog(ls LogsTestSuite, logFileName, content string, recurrence int) {
 	// Determine the OS and set the appropriate log path and command.
-	var cmd, checkCmd string
+	var checkCmd, logPath string
 	t := ls.T()
 	t.Helper()
 
 	var osStr string
+
+	logContent := strings.Repeat(content+"\n", recurrence)
+
 	switch ls.Env().RemoteHost.OSFamily {
 	case os.WindowsFamily:
 		osStr = "windows"
 		t.Log("Generating Windows log.")
-		cmd = fmt.Sprintf("echo %s > %s\\%s", strings.Repeat(content+" ", recurrence), WindowsLogsFolderPath, logFile)
-		checkCmd = fmt.Sprintf("Get-Content %s\\%s", WindowsLogsFolderPath, logFile)
+		// Unless a log line is newline terminated, the log agent will not pick it up. This is a known behavior.
```

Comment:
Maybe we should update this comment, because `\n` still means the log is newline terminated. It's just that windows uses `\r\n` instead.